Self-Consistent-Field/vcd.py

```diff
@@ -0,0 +1,874 @@
+#    symmetry c1
+#    units bohr
+#    noreorient
+#""")
+#mol = psi4.geometry("""
+#    O           -0.0000000000        1.3127774000       -0.1092016150
+#    O           -0.0000000000       -1.3127774000       -0.1092016150
+#    H            1.4385007800        1.7079033100        0.8736129170
+#    H           -1.4385007800       -1.7079033100        0.8736129170
+#    symmetry c1
+#    units bohr
+#    noreorient
+#""")
+mol = psi4.geometry("""
+    O       0.0000000000        1.3192641900       -0.0952542913
+    O      -0.0000000000       -1.3192641900       -0.0952542913
+    H       1.6464858700        1.6841036400        0.7620343300
+    H      -1.6464858700       -1.6841036400        0.7620343300
+    symmetry c1
+    units bohr
+    noreorient
+    no_com
+""")
+
+psi4.core.set_active_molecule(mol)
+
+geom = psi4.core.Matrix.to_array(mol.geometry())
+
+# Set Psi4 Options
+options = {'BASIS':'STO-3G',
+           'SCF_TYPE':'PK',
+           'E_CONVERGENCE':1e-10,
+           'D_CONVERGENCE':1e-10}
+psi4.set_options(options)
+
+# Perform SCF Energy Calculation
+rhf_e, wfn = psi4.frequency('SCF', return_wfn=True)
+
+# Relevant Variables
+natoms = mol.natom()
+nmo = wfn.nmo()
+nocc = wfn.doccpi()[0]
+nvir = nmo - nocc
+
+# MO Coefficients
+C = wfn.Ca_subset("AO", "ALL")
+npC = psi4.core.Matrix.to_array(C)
+
+# Integral generation from Psi4's MintsHelper
+mints = psi4.core.MintsHelper(wfn.basisset())
+
+# Build T, V, and S
+T = mints.ao_kinetic()
+npT = psi4.core.Matrix.to_array(T)
+V = mints.ao_potential()
+npV = psi4.core.Matrix.to_array(V)
+S = mints.ao_overlap()
+npS = psi4.core.Matrix.to_array(S)
+
+# Transform S to MO Basis
+npS_mo = 2.0 * np.einsum('uj,vi,uv', npC, npC, npS, optimize=True)
+
+# Build AO Dipole Integrals
+MU = mints.ao_dipole()
+AM = mints.ao_angular_momentum()
+npMU = []
+npAM = []
+npMU_mo = []
+npAM_mo = []
+for cart in range(3):
+    npMU.append(psi4.core.Matrix.to_array(MU[cart]))
+    npAM.append(psi4.core.Matrix.to_array(AM[cart]))
+    npAM[cart] *= -0.5
+    npMU_mo.append(np.einsum('uj,vi,uv', npC, npC, npMU[cart], optimize=True))
+    npAM_mo.append(np.einsum('uj,vi,uv', npC, npC, npAM[cart], optimize=True))
+
+# Build ERIs
+ERI = mints.mo_eri(C, C, C, C)
+npERI = psi4.core.Matrix.to_array(ERI)
+# Physicist notation
+npERI = npERI.swapaxes(1, 2)
+
+# Build Core Hamiltonian in AO basis
+H_core = npT + npV
+
+# Transform H to MO basis
+H = np.einsum('uj,vi,uv', npC, npC, H_core, optimize=True)
+
+# Build Fock Matrix
+F = H + 2.0 * np.einsum('pmqm->pq', npERI[:, :nocc, :, :nocc], optimize=True)
+F -= np.einsum('pmmq->pq', npERI[:, :nocc, :nocc, :], optimize=True)
+
+# Occupied and Virtual Orbital Energies
+F_occ = np.diag(F)[:nocc]
+F_vir = np.diag(F)[nocc:nmo]
+
+# Build Reference OPDM
+npD_ao = 2.0 * np.einsum('ui,vi->uv', npC[:, :nocc], npC[:, :nocc], optimize=True)
+# Transform to MO Basis
+ref_opdm = np.einsum('iu,uv,vw,wx,xj', npC.T, npS.T, npD_ao, npS, npC, optimize=True)
```

Comment:
The reference 1RDM in MO basis is the identity in the OCC-OCC subspace and zeros everywhere else. You don't need to do an AO-to-MO transformation.

Reply:
Of course. I fixed it to avoid unnecessary transformation.